### PR TITLE
Create default.bpei0.toml

### PR DESCRIPTION
**Pull request by Smig0l**
Added the toml file with addresses valid for BPEI (Pokemon Emerald Italian Version). Some are sourced from [this other abandoned project](https://github.com/Gamer2020/PokemonGameEditor/blob/master/roms.ini).
Currently i am adding more values as i discover them.

### CODE REVIEW

**Comment by haven1433** [comment]
When you're ready, you'll want to add `BPEI0` to the list in the constructor of `HardCodeTablesModel.cs` so that HMA knows to look for a default.*.toml file for Italian Emerald.

**Comment by Smig0l** [comment]
i opened the project in visual studio and fixed an exception (basically the only address that was missing was `data.pokemon.types.names`).
now the maps view works!!

File: src/HexManiac.Core/Models/Code/default.bpei0.toml [hunk 1]
```diff
@@ -0,0 +1,2858 @@
...
+Format = '''`asc`4'''
+DefaultHash = ''''''
+
+[[NamedAnchors]]
+Name = '''data.header.makercode'''
+Address = 0x0000B0
+Format = '''`asc`2'''
+DefaultHash = ''''''
+
+#[[NamedAnchors]]
```
Comment (haven1433) [inline src/HexManiac.Core/Models/Code/default.bpei0.toml:27]: Please remove all the commented out lines (starting with `#`) before we integrate the change. So we can more easily see just the important parts of the .toml file.

**Comment by haven1433** [comment]
> even by adding BPEI0 to HardCodeTablesModel.cs the rom isn't recognized and so the .toml isn't loaded

I went and checked how it worked for the French roms, it's just the code in `HardCodeTablesModel.cs`. I don't have access to an Italian rom, so I'll be depending on you to test that it works. You should be able to see a marked difference when first loading a the rom between when it works and when it doesn't.

**Comment by Smig0l** [comment]
> > even by adding BPEI0 to HardCodeTablesModel.cs the rom isn't recognized and so the .toml isn't loaded
> 
> I went and checked how it worked for the French roms, it's just the code in `HardCodeTablesModel.cs`. I don't have access to an Italian rom, so I'll be depending on you to test that it works. You should be able to see a marked difference when first loading a the rom between when it works and when it doesn't.

ok, so from what i have tested so far, pasting manually default.bpei0.toml under the "Resources" folder make HMA recognize it and populate the .toml with correct values... but obviously when building the exe it should put it automatically but it doesn't...
if you want to test yourself [here](https://github.com/Smig0l/GBA_stuff/blob/main/Pokemon%20Versione%20Smeraldo%20ita.gba) is the italian rom.

**Comment by haven1433** [comment]
> when building the exe it should put it automatically but it doesn't...

Oh! That's in the WPF .csproj file. You just need to tell it to copy the file to the resource directory, you can look how the other files are being copied.

**Comment by Smig0l** [comment]
> > when building the exe it should put it automatically but it doesn't...
> 
> Oh! That's in the WPF .csproj file. You just need to tell it to copy the file to the resource directory, you can look how the other files are being copied.

image
image
i see but even if it tells copy *.toml it doesn't works

**Comment by haven1433** [comment]
https://github.com/haven1433/HexManiacAdvance/blob/216dece90b4cd5ae04181a4031b3c6151a9e4960/src/HexManiac.Core/HexManiac.Core.csproj#L52 this is what I was thinking of, I was telling you the wrong csproj file

**Comment by Smig0l** [comment]
> https://github.com/haven1433/HexManiacAdvance/blob/216dece90b4cd5ae04181a4031b3c6151a9e4960/src/HexManiac.Core/HexManiac.Core.csproj#L52
> 
> this is what I was thinking of, I was telling you the wrong csproj file

Got it! thanks! now it copies it

File: src/HexManiac.Core/HexManiac.Core.csproj [hunk 2]
```diff
@@ -43,6 +43,9 @@
     <None Include="Models\Code\default.bpre1.toml">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+    <None Include="Models\Code\default.bpei0.toml">
```
Comment (haven1433) [inline src/HexManiac.Core/HexManiac.Core.csproj:46]: Replace the tabs with spaces to make everything line up nice.

Comment (haven1433) [reply src/HexManiac.Core/HexManiac.Core.csproj:46]: I looked through everything else, this is my last request :) almost ready to pull!

Comment (Smig0l) [reply src/HexManiac.Core/HexManiac.Core.csproj:47]: Sorry was on mobile and made a mess😳😳

**Pull request merged**